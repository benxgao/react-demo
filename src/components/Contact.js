import React, { PropTypes } from 'react';

const Contact = ({
  classes,
  contact,
}) => {
  const company = contact.get('company').get('name');
  const email = contact.get('email');
  const name = contact.get('name');
  const phone = contact.get('phone');
  const website = contact.get('website');
  const slogan = contact.get('company').get('bs');

  return (
    <div className={classes}>
      <div className="namecard">
        <div className="namecard__header">
          <div className="namecard__header-name">{name}</div>
          <div className="namecard__header-subtitle">{company}</div>
        </div>
        <div className="namecard__body">
          <div className="namecard__body-wrapper">
            <div className="namecard__item namecard__item--head">{slogan}</div>
            <div className="namecard__item"><b>Phone:</b> {phone}</div>
            <div className="namecard__item"><b>E-mail:</b> {email}</div>
            <div className="namecard__item"><b>URL:</b> {website} </div>
          </div>
        </div>
      </div>
    </div>
  );
};

Contact.propTypes = {
  classes: PropTypes.string.isRequired,
  contact: PropTypes.object.isRequired,
};

Contact.displayName = 'Contact';

export default Contact;
