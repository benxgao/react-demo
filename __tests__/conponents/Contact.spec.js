import Immutable from 'immutable';
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Contact from '../../src/components/Contact';
import dataMock from '../../__mocks__/dataMock';

const propsStub = {
  classes: 'contact css classnames',
  contact: Immutable.fromJS(dataMock[0]),
};

describe('<Contact />', () => {
  describe('render()', () => {
    it('should render Contact list when isLoading is false', () => {
      const props = Object.assign({}, propsStub);
      const wrapper = shallow(
        <Contact {...props} />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
