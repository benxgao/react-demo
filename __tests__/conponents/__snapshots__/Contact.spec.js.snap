exports[`<Contact /> render() should render Contact list when isLoading is false 1`] = `
<div
  className="contact css classnames">
  <div
    className="namecard">
    <div
      className="namecard__header">
      <div
        className="namecard__header-name">
        Leanne Graham
      </div>
      <div
        className="namecard__header-subtitle">
        Romaguera-Crona
      </div>
    </div>
    <div
      className="namecard__body">
      <div
        className="namecard__body-wrapper">
        <div
          className="namecard__item namecard__item--head">
          harness real-time e-markets
        </div>
        <div
          className="namecard__item">
          <b>
            Phone:
          </b>
           
          1-770-736-8031 x56442
        </div>
        <div
          className="namecard__item">
          <b>
            E-mail:
          </b>
           
          Sincere@april.biz
        </div>
        <div
          className="namecard__item">
          <b>
            URL:
          </b>
           
          hildegard.org
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
